bionic/aip/future.py

```diff
@@ -0,0 +1,96 @@
+import blocks
+import logging
+import time
+import googleapiclient.discovery
+from concurrent.futures import Future as _Future, TimeoutError, CancelledError
+from enum import Enum, auto
+
+
+class AIPError(Exception):
+    pass
+
+
+class State(Enum):
+    STATE_UNSPECIFIED = auto()
+    QUEUED = auto()
+    PREPARING = auto()
+    RUNNING = auto()
+    SUCCEEDED = auto()
+    FAILED = auto()
+    CANCELLING = auto()
+    CANCELLED = auto()
+
+
+EXECUTE = {State.STATE_UNSPECIFIED, State.QUEUED, State.PREPARING, State.RUNNING}
+CANCEL = {State.CANCELLING, State.CANCELLED}
+FAIL = {State.FAILED}
+
+
+class Future(_Future):
```

Comment:
Can you also leave a breadcrumb for API documentation? I feel like I jumped through a few hoops to finally find [this documentation](https://cloud.google.com/ai-platform/training/docs/reference/rest/v1/projects.jobs#State).